test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1347,10 +1336,19 @@ private synchronized void startAndPublishNodesAndClients(List<NodeAndClient> nod
                 // special case for 1 node master - we can't update the min master nodes before we add more nodes.
                 updateMinMasterNodes(currentMasters + newMasters);
             }
-            for (NodeAndClient nodeAndClient : nodeAndClients) {
-                nodeAndClient.startNode();
-                publishNode(nodeAndClient);
+            List<Future<?>> futures = nodeAndClients.stream().map(node -> executor.submit(node::startNode)).collect(Collectors.toList());
+            try {
+                for (Future<?> future : futures) {
+                    future.get();
+                }
+            } catch (InterruptedException e) {
+                Thread.interrupted();
+                return;
```

Comment:
The interrupted exception is always a battle as to what to do. I never seen it in practice but I have to deal with it because of checked exceptions. Not though that I didn't swallow it but rather intended to set the thread flag (which I failed miserably to do so and will fix). The alternative is to throw a `RuntimeException but that feels ugly as well (and I didn't want to force everyone to deal with InterruptedException by adding it to the signature). Which option do you prefer?